src/cocotb/share/lib/vpi/VpiImpl.cpp

```diff
@@ -344,7 +344,8 @@ GpiObjHdl *VpiImpl::native_check_create(const std::string &name,
         }
 
         for (auto rgn = vpi_scan(iter); rgn != NULL; rgn = vpi_scan(iter)) {
-            if (vpi_get(vpiType, rgn) == vpiGenScope) {
+            auto rgn_type = vpi_get(vpiType, rgn);
+            if (rgn_type == vpiGenScope || rgn_type == vpiModule) {
```

Comment:
This actually enabled Icarus to pick up Interface arrays by name now.  Before it would fail on test_name_matches_iter